app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt

```diff
@@ -243,12 +245,21 @@ class NewProfileFragment : Fragment() {
         var etime = "End Time: ${profile.ehr}:${profile.emin}"
         val profileData = workDataOf(Pair("ActiveProfileId", profile.profileId), Pair("Profile_Name", profile.name), (Pair("VibrateKey", profile.vibSwitch)), (Pair("EndTimeKey", etime)))
 
-        val startAlarmRequest = OneTimeWorkRequest.Builder(StartAlarm::class.java)
-            .addTag(profile.profileId.toString())
-            .setInputData(profileData)
-            .setInitialDelay(c.timeInMillis - System.currentTimeMillis(), TimeUnit.MILLISECONDS)
-            .build()
-        context?.let { WorkManager.getInstance(it).enqueue(startAlarmRequest) }
+        if (repeatWeeklySwitch.isChecked) {
+            val startAlarmRequest = PeriodicWorkRequest.Builder(StartAlarm::class.java, 7, TimeUnit.DAYS)
```

Comment:
A little code improvement change.
val startAlarmRequest =   if (repeatWeeklySwitch.isChecked) {.......
and take context?.let { WorkManager.getInstance(it).enqueue(startAlarmRequest) } out of if block.